src/api/FileActivities.js

```diff
@@ -0,0 +1,166 @@
+
+    data.map(item => {
+        if (item.source) {
+            // UAA follows a lowercased enum naming convention, convert to uppercase to align with task api
+            const taskItem = item.source[FILE_ACTIVITY_TYPE_TASK];
+            const commentItem = item.source[FILE_ACTIVITY_TYPE_COMMENT];
+            if (taskItem) {
+                if (!!taskItem.assigned_to && !!taskItem.assigned_to.entries) {
+                    taskItem.assigned_to.entries.map(entry => {
+                        entry.role = entry.role.toUpperCase();
+                        entry.status = entry.status.toUpperCase();
+
+                        return entry;
+                    });
+                }
+                if (taskItem.completion_rule && isString(taskItem.completion_rule)) {
+                    taskItem.completion_rule = taskItem.completion_rule.toUpperCase();
+                }
+                if (taskItem.status && isString(taskItem.status)) {
+                    taskItem.status = taskItem.status.toUpperCase();
+                }
+                if (taskItem.task_type && isString(taskItem.task_type)) {
+                    taskItem.task_type = taskItem.task_type.toUpperCase();
+                }
+
+                taskItem.created_by = {
+                    target: taskItem.created_by,
+                };
+            } else if (commentItem) {
+                commentItem.tagged_message = commentItem.tagged_message || commentItem.message;
+
+                if (commentItem.replies) {
+                    commentItem.replies.map(reply => {
+                        reply.tagged_message = reply.tagged_message || reply.message || '';
+
+                        return reply;
+                    });
+                }
+            }
+
+            parsedData.push(...Object.values(item.source));
+        }
+
+        return item;
+    });
+
+    return { entries: parsedData };
+};
+
+const getFileActivityQueryParams = (
+    fileID: string,
+    activityTypes?: FileActivityTypes[],
+    enableReplies?: boolean = true,
+) => {
+    const baseEndpoint = `/file_activities?file_id=${fileID}`;
+
+    return `${baseEndpoint}${
+        !!activityTypes && !!activityTypes.length ? `&activity_types=${activityTypes.join()}` : ''
+    }${enableReplies ? '&enable_replies=true&reply_limit=1' : ''}`;
+};
+
+class FileActivities extends Base {
+    /**
+     * API URL for file activities
+     *
+     * @param {string} [id] - a box file id
+     * @return {string} base url for files
+     */
+    getUrl(id?: string): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getBaseApiUrl()}${getFileActivityQueryParams(id, undefined, true)}`;
+    }
+
+    /**
+     * API URL for filtered file activities
+     *
+     * @param {string} [id] - a box file id
+     * @return {string} base url for files
+     */
+    getFilteredUrl(id?: string, activityTypes?: FileActivityTypes[]): string {
```

Comment:
We don't, however I did encounter a linting error when trying to add `activityTypes` params to `getUrl` since the `Base` component is expecting 1 parameter for `getUrl`. Thats why I created a separate method.